app/routes.js

```diff
@@ -1,27 +1,21 @@
-const { Router } = require('express')
+const {Router} = require('express')
 const Joi = require('@hapi/joi')
 const routes = Router()
+const ValidationMiddleware = require('./validation_middleware')
 const UserController = require('./users/user.controller')
-const HttpCodes = require('./httpcodes')
 
 const userSchema = Joi.object({
-  userName: Joi.string()
-    .alphanum()
-    .required()
+    userName: Joi.string()
+        .alphanum()
+        .required()
 })
 
-routes.post('/users', (request, response) => {
-  const { error } = userSchema.validate(request.body)
-  if (error != null) {
-    response.status(HttpCodes.BAD_REQUEST)
-    response.send(error.details.map(i => i.message).join(','))
-  } else {
+routes.post('/users', ValidationMiddleware(userSchema), (request, response) => {
```

Comment:
For the controller function you could directly pass it as a parameter, since functions are treated like objects or any other type in javascript. I.e. `routes.post('/users/, ValidationMiddleware(userSchema), UserController.createUser)`

Reply:
🤯  That is pretty cool